nuxeo-mqueues-core/src/main/java/org/nuxeo/ecm/platform/importer/mqueues/mqueues/chronicle/ChronicleMQueue.java

```diff
@@ -34,30 +44,44 @@
 import java.nio.file.Path;
 import java.time.Duration;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
 import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Stream;
 
-import static net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder.binary;
-
 /**
- * Chronicle Queue implementation of MQueue.
- *
- * Note that for performance reason the class loader assertion are disabled.
+ * Chronicle Queue implementation of MQueue. Note that for performance reason the class loader assertion are disabled.
  *
  * @since 9.1
  */
-public class ChronicleMQueue<M extends Externalizable> implements MQueue<M> {
+public class ChronicleMQueue<M extends Externalizable> implements MQueue<M>, StoreFileListener {
+
     private static final Log log = LogFactory.getLog(ChronicleMQueue.class);
+
+    private static final String SECOND_ROLLING_PERIOD = "s";
+
+    private static final String MINUTE_ROLLING_PERIOD = "m";
+
+    private static final String HOUR_ROLLING_PERIOD = "h";
+
+    private static final String DAY_ROLLING_PERIOD = "d";
+
     private static final String QUEUE_PREFIX = "Q-";
+
     private static final int POLL_INTERVAL_MS = 100;
 
     private final List<ChronicleQueue> queues;
+
     private final int nbQueues;
+
     private final File basePath;
+
     private final String name;
 
+    private int retentionNbCycles;
```

Comment:
Never `private`.
@bdelbosc please do a code review to use `protected` or `public` instead.